client/src/components/cardClosedTrade.js

```diff
@@ -45,14 +45,15 @@ class CardClosedTrade extends Component {
             color: this.props.color ? this.props.color : "blue",
             bookHave: this.props.bookHave ? this.props.bookHave : "History of India",
             bookWant: this.props.bookWant ? this.props.bookWant : "History of India"
-        }
-        this.printMessage = this.printMessage.bind(this)
-        this.renderCenter = this.renderCenter.bind(this)
+        };
+        this.printMessage = this.printMessage.bind(this);
+        this.renderCenter = this.renderCenter.bind(this);
+        this.onClickHandler = this.onClickHandler.bind(this);
     }
     
     printMessage(){
-        var type = this.props.color
-        if (type === 'P'){
+        var type = this.props.color;
```

Comment:
You are using es6 with classes, so move to block scoped declarations of const and let